lib/jekyll/collection.rb

```diff
@@ -100,15 +100,15 @@ def filtered_entries
     # Returns a String containing the directory name where the collection
     #   is stored on the filesystem.
     def relative_directory
-      @relative_directory ||= "_#{label}"
+      @relative_directory ||= Pathname.new(directory).relative_path_from(Pathname.new(site.source)).to_s
```

Comment:
It'd be a breaking change if plugins did `instance_variable_set(:@relative_directory, "blah")` and hoped it would affect `directory`. It reverses their role for sure, but instance variable manipulation like that is 💯 % not support.